src/test/scala/com/microsoft/hyperspace/index/HybridScanTest.scala

```diff
@@ -0,0 +1,138 @@
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.{DataFrame, QueryTest}
+import org.apache.spark.sql.execution.datasources._
+
+import com.microsoft.hyperspace.{Hyperspace, SampleData}
+import com.microsoft.hyperspace.index.Content.Directory.FileInfo
+import com.microsoft.hyperspace.index.plans.logical.BucketUnion
+import com.microsoft.hyperspace.index.rules.{FilterIndexRule, JoinIndexRule}
+import com.microsoft.hyperspace.util.FileUtils
+
+class HybridScanTest extends QueryTest with HyperspaceSuite {
+  override val systemPath = new Path("src/test/resources/hybridScanTest")
+
+  private val sampleData = SampleData.testData
+  private val sampleParquetDataLocation = "src/test/resources/sampleparquet2"
+  private var df: DataFrame = _
+  private var hyperspace: Hyperspace = _
+  private var files: Seq[FileInfo] = _
+  private val indexConfig1 = IndexConfig("index1", Seq("clicks"), Seq("query"))
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    val sparkSession = spark
+    import sparkSession.implicits._
+    hyperspace = new Hyperspace(spark)
+    FileUtils.delete(new Path(sampleParquetDataLocation))
+    val dfFromSample = sampleData.toDF("Date", "RGUID", "Query", "imprs", "clicks")
+    dfFromSample.write.parquet(sampleParquetDataLocation)
+
+    df = spark.read.parquet(sampleParquetDataLocation)
+    hyperspace.createIndex(df, indexConfig1)
+
+    files = df.queryExecution.optimizedPlan.collect {
+      case LogicalRelation(
+          HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+          _,
+          _,
+          _) =>
+        location.allFiles.map(FileInfo(_))
+    }.flatten
+    val sourcePath = new Path(files.head.name)
+    val destPath = new Path(files.head.name + ".copy")
+    sourcePath.getFileSystem(new Configuration).copyToLocalFile(sourcePath, destPath)
+  }
+
+  before {}
+
+  after {}
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+    spark.sessionState.conf.unsetConf(IndexConstants.INDEX_HYBRID_SCAN_ENABLED)
+    spark.sessionState.conf.unsetConf("spark.sql.autoBroadcastJoinThreshold")
+    FileUtils.delete(new Path(sampleParquetDataLocation))
+  }
+
+  test("HybridScan filter rule test") {
+    spark.conf.set("spark.hyperspace.index.hybridscan.enabled", false)
+
+    df = spark.read.parquet(sampleParquetDataLocation)
+    val query = df.filter(df("clicks") <= 2000).select(df("query"))
+    val transformed = FilterIndexRule(query.queryExecution.optimizedPlan)
+    assert(transformed.equals(query.queryExecution.optimizedPlan), "No plan transformation.")
+
+    spark.conf.set("spark.hyperspace.index.hybridscan.enabled", true)
+    val planWithHybridScan = FilterIndexRule(query.queryExecution.optimizedPlan)
+    assert(
+      !planWithHybridScan.equals(query.queryExecution.optimizedPlan),
+      "Plan should be transformed.")
+
+    // check appended file is added to relation node or not
+    var relationCnt = 0
+    planWithHybridScan foreach {
```

Comment:
I would just use `collect` instead of `relationCnt`:
```
val relations = planWithHybridScan.collect {}
assert(relations.length == 1)
```

Reply:
Thanks! But besides that I'd like to check if the file path is included in the relation or not. :)

Reply:
You can still do that inside `collect`, no?

Reply:
Thanks for the tip 👍